File: src/player-pose.js
import { advancePhysicalPose } from "./physical-pose.js";
import { startInjury } from "./features.js";

// The fall -5 terminator (0x10d44) stops movement and resolves injury in
// the player's animation pass, rather than waiting for the next timer tick.
export function advancePlayerPose(s, i, dt) {
  const p = s.players[i];
  if (!advancePhysicalPose(p, i, s.period, dt)) return false;
  p.stun = p.actionTime = 0;
  p.moveX = p.moveZ = p.fallX = p.fallZ = 0;
  p.fallAttack = p.fallAttackTime = 0;
  p.fallFinishing = p.jumping = false;
  if (p.health <= 0) return startInjury(s, i);
  // Healthy -5 selects anim_lying_down and clears its cursor at 0x10d84.
  p.poseCursor = 0;
  p.action = 0;
  return false;
}

File: server/player_pose.go
package main

// The fall -5 terminator resolves movement and injury in the animation pass.
func (s *State) advancePlayerPose(i int, dt float64) bool {
	p := &s.Players[i]
	if !advancePhysicalPose(p, i, s.Period, dt) {
		return false
	}
	p.Stun, p.ActionTime = 0, 0
	p.moveX, p.moveZ, p.fallX, p.fallZ = 0, 0, 0, 0
	p.fallAttack, p.fallAttackTime = 0, 0
	p.fallFinishing, p.jumping = false, false
	if p.Health <= 0 {
		return s.startInjury(i)
	}
	// Healthy -5 selects anim_lying_down and resets its cursor (0x10d84).
	p.poseCursor = 0
	p.Action = 0
	return false
}
